container/container.go

```diff
@@ -183,6 +185,34 @@ func (container *Container) WriteHostConfig() error {
 	return json.NewEncoder(f).Encode(&container.HostConfig)
 }
 
+// SetupWorkingDirectory sets up the container's working directory as set in container.Config.WorkingDir
+func (container *Container) SetupWorkingDirectory() error {
+	if container.Config.WorkingDir == "" {
+		return nil
+	}
+	container.Config.WorkingDir = filepath.Clean(container.Config.WorkingDir)
+
+	pth, err := container.GetResourcePath(container.Config.WorkingDir)
+	if err != nil {
+		return err
+	}
+
+	pthInfo, err := os.Stat(pth)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+
+		if err := system.MkdirAll(pth, 0755); err != nil {
+			return err
+		}
+	}
+	if pthInfo != nil && !pthInfo.IsDir() {
```

Comment:
Looks like this part was to make sure that the returned error code is readable by humans (ie. in relation to the container root rather than the host root).

I've added it back in in the case of an error creating the directory.